gdc_ng_models/models/batch.py

```diff
@@ -0,0 +1,149 @@
+            self.id,
+            self.name,
+            self.project_id,
+            self.status,
+            created_datetime,
+            updated_datetime,
+        )
+
+    def to_dict(self):
+        created_datetime = (
+            self.created_datetime.isoformat() if self.created_datetime else None
+        )
+        updated_datetime = (
+            self.updated_datetime.isoformat() if self.updated_datetime else None
+        )
+
+        return {
+            "id": self.id,
+            "name": self.name,
+            "project_id": self.project_id,
+            "status": self.status,
+            "created_datetime": created_datetime,
+            "updated_datetime": updated_datetime,
+        }
+
+    def to_json(self):
+        """Returns a JSON safe representation of a batch"""
+        return json.loads(json.dumps(self.to_dict()))
+
+
+class BatchMembership(Base, audit.AuditColumnsMixin):
+    """Membership class to represent which nodes belong in a batch.
+
+    Attributes:
+        batch_id: id of the batch
+        node_id: id of the node
+        created_datetime: the date and time when node was added to batch
+        updated_datetime: the date and time when node membership was last updated
+    """
+
+    __tablename__ = "batch_membership"
+    __table_args__ = (
+        schema.PrimaryKeyConstraint("batch_id", "node_id", name="batch_membership_pk"),
+        schema.ForeignKeyConstraint(
+            ("batch_id",),
+            ("batch.id",),
+            name="batch_membership_batch_id_fk",
+        ),
+    )
+
+    batch_id = sqlalchemy.Column(sqlalchemy.BigInteger, nullable=False)
+    node_id = sqlalchemy.Column(sqlalchemy.Text, nullable=False)
+
+    batch = orm.relationship("Batch", back_populates="members")
+
+    def __repr__(self):
+        created_datetime = (
+            self.created_datetime.isoformat() if self.created_datetime else None
+        )
+        updated_datetime = (
+            self.updated_datetime.isoformat() if self.updated_datetime else None
+        )
+
+        return "<BatchMembership(batch_id='{}', node_id='{}', created_datetime='{}', updated_datetime='{}')>".format(
+            self.batch_id,
+            self.node_id,
+            created_datetime,
+            updated_datetime,
+        )
+
+    def to_dict(self):
+        created_datetime = (
+            self.created_datetime.isoformat() if self.created_datetime else None
+        )
+        updated_datetime = (
+            self.updated_datetime.isoformat() if self.updated_datetime else None
+        )
+
+        return {
+            "batch_id": self.batch_id,
+            "node_id": self.node_id,
+            "created_datetime": created_datetime,
+            "updated_datetime": updated_datetime,
+        }
+
+    def to_json(self):
+        """Returns a JSON safe representation of a membership object"""
+        return json.loads(json.dumps(self.to_dict()))
```

Comment:
another missing newline end-of-file, maybe we need to add another hook for the precommit, but definitely out of scope